packages/connections/src/components/connections.tsx

```diff
@@ -48,121 +64,25 @@ const formContainerStyles = css({
   },
 });
 
-function getDefaultConnectionInfo() {
-  return {
-    connectionOptions: {
-      connectionString: 'mongodb://localhost:27017',
-    },
-  };
-}
-
-const mockRecents: ConnectionInfo[] = [];
-for (let i = 0; i < 15; i++) {
-  mockRecents.push({
-    id: `mock-connection-${i}`,
-    connectionOptions: {
-      connectionString: `mongodb://localhost:2${
-        5000 + Math.floor(Math.random() * 5000)
-      }`,
-    },
-    lastUsed: new Date(Date.now() - (Date.now() / 2) * Math.random()),
-  });
-}
-
-const mockConnections = [
-  {
-    id: 'mock-connection-dev',
-    connectionOptions: {
-      connectionString: '',
-    },
-    favorite: {
-      name: 'Development cluster',
-      color: '#deb342',
-    },
-    lastUsed: new Date(),
-  },
-  {
-    id: 'mock-connection-atlas',
-    connectionOptions: {
-      connectionString:
-        'mongodb+srv://testUserForTesting:notMyRealPassword@test.mongodb.net/test?authSource=admin&replicaSet=art-dev-shard-0&readPreference=primary&ssl=true',
-    },
-    favorite: {
-      name: 'Atlas test',
-      color: '#d4366e',
-    },
-    lastUsed: new Date(),
-  },
-  {
-    id: 'mock-connection-empty-connection',
-    connectionOptions: {
-      connectionString: '',
-    },
-    favorite: {
-      name: 'super long favorite name - super long favorite name - super long favorite name - super long favorite name',
-      color: '#5fc86e',
-    },
-    lastUsed: new Date(),
-  },
-  {
-    id: 'mock-connection-invalid string',
-    connectionOptions: {
-      connectionString: 'invalid connection string',
+function Connections({
+  appRegistry,
+}: {
+  appRegistry: AppRegistry;
+}): React.ReactElement {
+  const [
+    {
+      activeConnectionId,
+      activeConnectionInfo,
+      connectingStatusText,
+      connectionAttempt,
+      connections,
+      isConnected,
     },
-    lastUsed: new Date(),
-  },
-  ...mockRecents,
-];
-const connections = mockConnections;
-
-type State = {
-  activeConnectionId?: string;
-  activeConnectionInfo: ConnectionInfo;
-};
-
-type Action =
-  | {
-      type: 'new-connection';
-      newConnectionId: string;
-    }
-  | {
-      type: 'set-active-connection';
-      connectionId: string;
-      connectionInfo: ConnectionInfo;
-    };
-
-function reducer(state: State, action: Action): State {
-  switch (action.type) {
-    case 'set-active-connection':
-      return {
-        ...state,
-        activeConnectionId: action.connectionId,
-        activeConnectionInfo: action.connectionInfo,
-      };
-    case 'new-connection':
-      return {
-        ...state,
-        activeConnectionId: action.newConnectionId,
-        activeConnectionInfo: {
-          ...getDefaultConnectionInfo(),
-          id: action.newConnectionId,
-        },
-      };
-    default:
-      return state;
-  }
-}
+    dispatch,
+  ] = useReducer(connectionsReducer, defaultConnectionsState());
```

Comment:
If reducer is complex enough that you're moving it to it's own place a common practice that I see applied sometimes is to move the whole thing out, including dispatches and other methods and have some sort of `useConnections()` hook with an actions-like API instead of exposing every internal detail of the state separately, so e.g., instead of manually dispatching 'new-connection' with all the required attributes, you can have something like:

```js
function useConnections() {
  // ...
  return {
    // ...
    createNewConnection() {
      dispatch({
        type: 'new-connection',
        connectionInfo: createNewConnectionInfo(),
      });
    }
  }
}

const { state, createNewConnection } = useConnections();

// ...

createNewConnection()
```

Combine this with the fact that most if not all of the appRegistry usage here is inside these state changing "actions" and that appRegistry is available through context, you don't need to leak any internal implementation details to your component outside of the hook.

Not suggesting you rewrite all this, maybe something that would come in handy one day as an approach.

Reply:
I like this, will do some updating and ping when updated. Would be good to do it now since with some of the future tickets they'll benefit from things being a bit more organized.